iceoryx_hoofs/include/iceoryx_hoofs/internal/posix_wrapper/thread.inl

```diff
@@ -0,0 +1,40 @@
+#ifndef IOX_HOOFS_POSIX_WRAPPER_THREAD_INL
+#define IOX_HOOFS_POSIX_WRAPPER_THREAD_INL
+
+#include "iceoryx_hoofs/posix_wrapper/thread.hpp"
+
+namespace iox
+{
+namespace posix
+{
+inline cxx::expected<ThreadError> ThreadBuilder::create(cxx::optional<Thread>& uninitializedThread,
+                                                        const cxx::function<void()>& callable) noexcept
+{
+    if (!callable)
+    {
+        std::cerr << "The thread cannot be created with an empty callable." << std::endl;
+        return cxx::error<ThreadError>(ThreadError::EMPTY_CALLABLE);
+    }
+
+    uninitializedThread.emplace();
+    uninitializedThread->m_callable = callable;
+
+    // create thread
+    auto createResult =
+        posixCall(pthread_create)(
+            &uninitializedThread->m_threadHandle, nullptr, Thread::startRoutine, &uninitializedThread->m_callable)
+            .successReturnValue(0)
+            .evaluate();
+    if (createResult.has_error())
+    {
+        uninitializedThread->m_isJoinable = false;
```

Comment:
Could you write this line like `uninitializedThread->m_isJoinable = !createResult.has_error();` and move it above the condition.
Also I would rename the variable to something like `m_isThreadConstructed`. The reason is that we do not distinguish joinable and detached threads with this variable but successfully and failed threads.